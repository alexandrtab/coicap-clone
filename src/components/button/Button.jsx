export const Button = ({
	children,
	type,
	onClickButton,
	isSubmit,
	className,
}) => (
	<>
		{isSubmit ? (
			<input
				value={ children }
				type={ type }
				className={ "btn" + " " + className }
				onSubmit={ onClickButton }
				/>
			) : (
				<button
					className={ "btn" + " " + className }
					type={ type }
					onClick={ onClickButton }
				>
					{children}
				</button>
			)}
	</>
	);
