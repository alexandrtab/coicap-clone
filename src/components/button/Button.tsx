import React from "react";

import { IButtonProps } from "./types";

export const Button: React.FC<IButtonProps> = ({
	text,
	type,
	onClickButton,
	isSubmit,
	className,
	children
}) => (
	<>
		{isSubmit ? (
			<input
				value={ text }
				type={ type }
				className={ "btn" + " " + className }
				onSubmit={ onClickButton }
			/>
		) : (
			<button
				className={ "btn" + " " + className }
				type={ type }
				onClick={ onClickButton }
			>
				{children}
				{text}
			</button>
		)}
	</>
);
