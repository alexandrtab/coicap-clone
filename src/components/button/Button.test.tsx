import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import { Button } from "./Button";

describe("Button", () => {
	render(
		<Button
			text= "button"
			type= "button"
			onClickButton={ () => {} }
			isSubmit={ false }
			className={ "" }
		/>
	);
	test("should render with text", () => {
		const input = screen.queryByRole("input");
		const button = screen.getByRole("button");

		expect(input).not.toBeInTheDocument();
		expect(button).toBeInTheDocument();
		expect(button).toMatchSnapshot();
		expect(input).toMatchSnapshot();

	});
});
