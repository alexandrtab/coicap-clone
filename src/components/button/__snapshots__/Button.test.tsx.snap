// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button 1`] = `
<button
  class="btn "
  type="button"
>
  button
</button>
`;

exports[`Button should render button 2`] = `null`;

exports[`Button should render with text 1`] = `
<button
  class="btn "
  type="button"
>
  button
</button>
`;

exports[`Button should render with text 2`] = `null`;
